File: src/main/java/dev/jarand/authprotectedrequests/authapi/AuthApiClientMock.java
package dev.jarand.authprotectedrequests.authapi;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Service;

import java.security.PublicKey;

@Service
@ConditionalOnProperty(name = "authentication.mock.enabled", havingValue = "true")
public class AuthApiClientMock implements AuthApiClient {

    @Override
    public PublicKey fetchPublicKey() {
        throw new IllegalStateException("Not implemented");
    }

    @Override
    public String refreshToken(String refreshToken) {
        throw new IllegalStateException("Not implemented");
    }
}

File: src/main/kotlin/dev/jarand/authprotectedrequests/authapi/AuthApiClientMock.kt
package dev.jarand.authprotectedrequests.authapi

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service
import java.security.PublicKey

@Service
@ConditionalOnProperty(name = ["authentication.mock.enabled"], havingValue = "true")
class AuthApiClientMock : AuthApiClient {

    override fun fetchPublicKey(): PublicKey {
        throw IllegalStateException("Not implemented")
    }

    override fun refreshToken(refreshToken: String): String? {
        throw IllegalStateException("Not implemented")
    }
}
